python2/ltk/apicalls.py

```diff
@@ -243,6 +248,57 @@ def add_document(self, source_locale, file_name, project_id, title, **kwargs):
         except requests.exceptions.ConnectionError:
             self.handleError()
         return r
+    
+    def document_add_reference(self, document_id, reference):
+        """ adds reference material to a document """
+        try:
+            uri = (API_URI['reference'] % locals())
+            payload = {'id': document_id}
+            document = open(reference['file'], 'rb')
+            files = {'content': (reference['file'], document)}
+            if 'name' in reference and reference['name']:
+                payload.update({'name': reference['name']})
+            if 'description' in reference and reference['description']:
+                payload.update({'description': reference['description']})
+            r = requests.post(self.host + uri, headers=self.headers, data=payload, files=files)
```

Comment:
I'm just not sure about this code but I am sure you are right, but is it possible that r is not defined at the time of return? Will this cause an error in python?

Reply:
also consider using more descriptive variable names such as resp or response

Reply:
The variable names are just following the pattern that existed before me.  I figured I'd keep the pattern instead of refactoring everything.
As for r being undefined, it will cause errors in the calling functions, but all of the ways for r to be set to undefined would throw their own errors before then.